styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -72,26 +73,33 @@
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import okio.ByteString;
 
-public final class BackfillResource {
+public final class BackfillResource implements Closeable {
 
   static final String BASE = "/backfills";
   private static final String SCHEDULER_BASE_PATH = "/api/v0";
   private static final String UNKNOWN = "UNKNOWN";
   private static final String WAITING = "WAITING";
+  private static final int CONCURRENCY = 64;
 
   private final Storage storage;
   private final String schedulerServiceBaseUrl;
   private final WorkflowValidator workflowValidator;
 
+  private final ForkJoinPool forkJoinPool;
+
   public BackfillResource(String schedulerServiceBaseUrl, Storage storage,
       WorkflowValidator workflowValidator) {
     this.schedulerServiceBaseUrl = Objects.requireNonNull(schedulerServiceBaseUrl);
     this.storage = Objects.requireNonNull(storage);
     this.workflowValidator = Objects.requireNonNull(workflowValidator, "workflowValidator");
+    this.forkJoinPool = new ForkJoinPool(CONCURRENCY);
```

Comment:
We're starting to have a lot of bespoke thread pools with throughout the code base, each with life cycle management etc. Maybe we should start injecting a shared FJP from `StyxScheduler` instead and centralize setup and cleanup?

Reply:
Makes sense. We might actually need multiple FJPs for different components to get better throughput.